Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,63 @@
-function setAlarm() {}
+let countdownTimer;
+let flashInterval;
+
+function setAlarm() {
+  document.getElementById("stop").addEventListener("click", () => {
+    if (flashInterval) {
+      clearInterval(flashInterval);
+      flashInterval = null;
+      document.body.style.backgroundColor = "";
+    }
+  });
+
+  let alarmTime = Number(document.querySelector("input").value);
+
+  if (!alarmTime || isNaN(alarmTime) || alarmTime <= 0) {
+    alert("Please enter a valid positive number for the alarm time");
+    return;
+  }
+
+  let timeLeft = alarmTime;
+
+  function formatTime(time) {
+    let minutes = Math.floor(time / 60);
+    let seconds = time % 60;
+    return `${minutes.toString().padStart(2, "0")}:${seconds
+      .toString()
+      .padStart(2, "0")}`;
+  }
+
+  function updateTitle(time) {
```

Comment:
Is there a reason you're defining all of these functions inside `setAlarm`? It makes it a bit harder to work out what `setAlarm` is doing because there are a bunch of functions in between the code actually doing stuff. Generally we tend to define functions at the top-level of a file instead of nested, unless there's a reason not to.

Reply:
@illicitonion Thanks! I’ve moved the helper functions outside setAlarm to keep the main function focused and improve readability.